Cloud Native SIG - Update Pluggable Storage story statuses (#3702)
jenkins-infra/jenkins.io: content/sigs/cloud-native/pluggable-storage.adoc

Below you can find a summary of ongoing activities and their current status:

[frame="topbot",grid="all",options="header", cols="20%,50%,30%"]
|=========================================================
|Type / Status | Comment | Implementation(s)

| **Artifacts** +
  **(Available)**
| Fully delivered, with support for uploading artifacts directly from agents.
  Related JEPs: jep:202[].

  link:/doc/developer/extensions/jenkins-core/#artifactmanagerfactory[Extension point]

| plugin:artifact-manager-s3[Artifact Manager on S3],
  plugin:azure-artifact-manager[Azure Artifact Manager],
  link:/doc/developer/extensions/jenkins-core/#artifactmanagerfactory[<More implementations>]

| **Credentials** +
  **(Available)**
| Completed before the JEP process was introduced.

link:https://www.jenkins.io/doc/developer/extensions/credentials/#credentialsprovider[Extension point]
| plugin:kubernetes-credentials-provider[Kubernetes Credentials Provider],
  plugin:aws-secrets-manager-credentials-provider[AWS Secrets Manager Credentials Provider],
  https://www.jenkins.io/doc/developer/extensions/credentials/#credentialsprovider[<More implementations>]

| Build logs +
  (Preview / Paused)

> | The story is completed for Pipeline Log Storage, only Jenkins Pipeline job types are supported.

Reviewer: Not sure I would say JEP-210 is “completed”. In particular

* the change to make `sh` steps stream properly was reverted due to some minor issues and was never reapplied (it is forcibly enabled by `pipeline-cloudwatch-logs`)
* the log cursor in core (also used by Blue Ocean) is still a `long` rather than an opaque implementation-defined type, making it impossible to efficiently support streaming display from cloud services

Author: Right. "preview" then?

Author: ```suggestion
| Pipeline Log Storage API and reference implementations are available for preview, only Jenkins Pipeline job types are supported.
```

| **System logs** +
  **(Available)**
| Jenkins supports custom log appenders using standard `java.util.logging`
  link:https://tutorials.jenkov.com/java-logging/configuration.html[configuration options].
| plugin:syslog-logger[Syslog logger],
  non-Jenkins implementation

| Task logs +
  (Not started)
| Storage of system logs and various tasks (e.g. agent connection or SCM polling)
| N/A

| Configurations +
  (Paused)
| Largely replaced by the plugin:configuration-as-code[Configuration as Code] plugin
  which allows storing Jenkins configurations in SCM or other locations without a database.

  Related JEPs: jep:213[]
| N/A

| **Test Results** + 
  **(Preview)**
| Foundation API is integrated into plugin:junit[JUnit Plugin 1.32].
  Remaining changes are tracked as link:https://github.com/jenkinsci/junit-plugin/issues/142[junit #142]
| https://github.com/timja/jenkins-junit-postgresql-plugin[JUnit PostgreSQL Plugin]

| Code Coverage results +
  (Not started)
| Planned only for plugins based on plugin:code-coverage-api[Code Coverage API] which unifies the storage implementation.
  See _Runs_ for other coverage report types.
| N/A

> | Runs +

Reviewer: ```suggestion
| Builds +
```

Author: ```suggestion
| Builds/Runs +
```

We still need to fixate the glossary as we discussed with @megathaum a while ago

Author: Will incorporate changes. Right now Builds are referenced in https://www.jenkins.io/doc/book/glossary/

  (Not started)

> | Storage of full run records in an external database.

Reviewer: ```suggestion
| Storage of full build records in an external database.
```

>   Includes storing run data which is not listed 

Reviewer: ```suggestion
  Includes storing build data which is not otherwise listed (such as logs or test results).
```
